test/integration/job/job_test.go

```diff
@@ -1026,6 +1026,334 @@ func TestIndexedJob(t *testing.T) {
 	validateTerminatedPodsTrackingFinalizerMetric(t, 5)
 }
 
+// This test has 4 parts which execute in the following order:
+// 1) Create Indexed Job then succeed/fail some indexes
+// 2) Optional: scale up then succeed/fail some indexes
+// 3) Optional: scale down then succeed/fail some indexes
+// 4) Optional: scale back up again
+//
+// Parts 1-3 are only executed if a flag in the test case
+// indicates that they should do so. Therefore, each test
+// case can toggle them to test various scenarios involving
+// scaling up/down/scaling up again, etc.
+func TestElasticIndexedJob(t *testing.T) {
+	cases := map[string]struct {
+		// Initial job
+		initialParallelism                int
+		initialCompletions                int
+		initialIndexesToSucceed           []int
+		wantSucceededIndexesBeforeScaleUp string
+		// Scale up param
+		shouldScaleUp                    bool
+		scaledUpParallelism              int
+		scaledUpCompletions              int
+		makeIndexesSucceedAfterUpdate    []int
+		makeIndexesFailBeforeScaleUp     []int
+		wantSucceededAfterScaleUp        int
+		wantSucceededIndexesAfterScaleUp string
+		// Scale down params
+		shouldScaleDown                    bool
+		scaledDownParallelism              int
+		scaledDownCompletions              int
+		wantSucceededAfterScaleDown        int
+		wantSucceededIndexesAfterScaleDown string
+		wantFailed                         int
+		// 2nd scale up params
+		shouldDoSecondScaleUp                                      bool
+		secondScaleUpParallelism                                   int
+		secondScaleUpCompletions                                   int
+		makeIndexesSucceedAfterSecondScaleUp                       []int
+		wantSucceededAfterSecondScaleUpBeforeNewPodsSucceed        int
+		wantSucceededAfterSecondScaleUpAfterNewPodsSucceed         int
+		wantSucceededIndexesAfterSecondScaleUpBeforeNewPodsSucceed string
+		wantSucceededIndexesAfterSecondScaleUpAfterNewPodsSucceed  string
+		// Misc
+		enableFeatureGate bool
+		wantErr           bool
+	}{
+		"feature gate turned off, mutation not allowed": {
+			initialParallelism:  2,
+			initialCompletions:  2,
+			shouldScaleUp:       true,
+			scaledUpParallelism: 4,
+			scaledUpCompletions: 4,
+			wantErr:             true,
+		},
+		"initial completions != initial parallelism": {
+			initialParallelism:  2,
+			initialCompletions:  3,
+			shouldScaleUp:       true,
+			scaledUpParallelism: 4,
+			scaledUpCompletions: 4,
+			wantErr:             true,
+		},
+		"updated completions != updated parallelism": {
+			initialParallelism:  2,
+			initialCompletions:  2,
+			shouldScaleUp:       true,
+			scaledUpParallelism: 4,
+			scaledUpCompletions: 5,
+			wantErr:             true,
+		},
```

Comment:
I don't think the above two tests are needed, we already have this covered in the validation unit tests.